ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/ImmDB.hs

```diff
@@ -496,31 +495,34 @@ stream db registry blockComponent from to = runExceptT $ do
           ImmDB.WrongHashError slot _ _ -> slot
 
     -- | The ImmutableDB doesn't support an exclusive end bound, so we stop
-    -- the iterator when it reaches its exclusive end bound.
-    stopAt :: forall a.
-              StreamTo blk
-           -> ImmDB.Iterator (HeaderHash blk) m (HeaderHash blk, a)
-           -> ImmDB.Iterator (HeaderHash blk) m (HeaderHash blk, a)
+    -- the iterator when the next block is the exclusive end bound.
+    stopAt :: StreamTo blk
```

Comment:
Rename to `makeExclusive` because it is already stopping at the inclusive upper bound.

Reply:
Actually, this function is called in both cases: when the upper bound is exclusive *or* inclusive. It will just do nothing in the second case. So I think the name is fine after all.